f8a_worker/workers/mercator.py

```diff
@@ -271,3 +275,22 @@ def get_outermost_items(list_):
             sorted_list = [i for i in sorted_list if
                            len(i['path'].split(os.path.sep)) == outermost_len]
         return sorted_list
+
+
+def validate_utf_json(result):
+    """Remove non UTF-8 characters from description if present.
+
+    :param result: dict, Dictionary having details of task
+    return: a dictionary received as input having non UTF characters removed if present
+    """
+    try:
+        if result is not None and 'details' in result and isinstance(result['details'], list):
+            for i in range(len(result['details'])):
+                if 'description' in result['details'][i]:
+                    result['details'][i]['description'] = ''.join(
+                        x for x in result['details'][i]['description']
+                        if x in string.printable)
```

Comment:
List iteration shall be simplified as below,

```suggestion
            for detail in result['details']:
                if 'description' in detail:
                    detail['description'] = ''.join(
                        x for x in detail['description']
                        if x in string.printable)
```